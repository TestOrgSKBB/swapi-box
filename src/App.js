import React, { Component } from 'react';
// import { fetchMovies } from './apiCalls'
import { fetchData } from './apiCalls';

class App extends Component {

  constructor() {
    super();
    this.state = {
      movies: {},
    }
  }

  componentDidMount = () => {
    fetchData('https://swapi.co/api/films/')
    .then(data => {
      console.log(data);
      return data;
    })
    .then(movies => this.setState({ movies }))
    .catch(error => console.log(error));
  }
  
  render() {
    return (
      <h1>Fetch testing branch</h1>
    );
  };
};

export default App;
