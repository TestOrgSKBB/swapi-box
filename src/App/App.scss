* {
  font-family: 'Barlow Condensed', sans-serif;
}

main {
background-image: 
url('../images/star-wars-planets.jpg');
background-size: cover; 
overflow: scroll;
height: 100vh;
opacity: .7;
position: fixed;
width: 100%
}

h1 {
  color: #BFCAC3;
  text-align: center;
  margin-top: 2%;
}

body {
  background-image: 
  url('../images/stars.gif');
    background-size: cover;
}

.main__div--info {
  display: flex;
  flex-direction: column;
  font-size: 1.5em;
}

.user-info {
margin-top:3%;
font-size: .8em;
}

.button__sign-out {
  background: rgba(55, 51, 62, 0.4);
  height: 5vh;
  color: #919192;
  width: 236%;
  border: 1px solid #919192;
  border-radius: 30px;
  font-size: 1em;
  font-weight: bold;
  cursor: pointer;
}

.button__sign-out:hover {
  background: rgba(143, 171, 172, .7);
  border:2px solid rgba(32, 37, 40, .8);
  color: #963025;
}

.main__div {
  align-items: center;
  height: 20vh;
  background: rgba(51, 51, 51, .7);
  display: flex;
  justify-content: space-around;
  width: 100%;
}

@keyframes neon4 {
  from {
    box-shadow: 0 0 0px #B6FF00, 0 0 10px #B6FF00, 0 0 30px #fff, 0 0 30px #B6FF00, 0 0 40px #B6FF00, 0 0 60px #B6FF00, 0 0 80px #B6FF00, 0 0 100px #B6FF00;
  }
  to {
    box-shadow: 0 0 10px #fff, 0 0 10px #fff, 0 0 5px #fff, 0 0 20px rgb(255, 44, 44), 0 0 40px rgb(255, 44, 44), 0 0 40px rgb(255, 44, 44), 0 0 60px rgb(255, 44, 44), 0 0 80px rgb(255, 44, 44);
  }
}

form {
  -webkit-animation: neon4 1.5s ease-in-out infinite alternate;
  -moz-animation: neon4 1.5s ease-in-out infinite alternate;
  animation: neon4 1.5s ease-in-out infinite alternate;
  border-radius: 20px;
}


