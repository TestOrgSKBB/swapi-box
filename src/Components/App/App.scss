$dark-grey: rgba(32, 37, 40, .8);
$red-font: #963025;
$sign-out-btn: rgba(55, 51, 62, 0.4);
$sign-out-btn-hover: rgba(143, 171, 172, .7);
$solid-light-grey: #919192;

* {
  font-family: 'Barlow Condensed', sans-serif;
}

main {
background-image: 
url('../../images/star-wars-planets.jpg');
background-size: cover; 
overflow: scroll;
height: 100vh;
opacity: .7;
position: fixed;
width: 100%
}

h1 {
  color: #BFCAC3;
  text-align: center;
  margin-top: 2%;
}

body {
  background-image: 
  url('../../images/stars.gif');
    background-size: cover;
}

.main__div--info {
  display: flex;
  flex-direction: column;
  font-size: 1.5em;
  margin-right: 47%;
}

.user-info {
margin-top:3%;
font-size: .8em;
}

.button__sign-out {
  background: $sign-out-btn;
  height: 5vh;
  color: $solid-light-grey;
  width: 236%;
  border: 1px solid $solid-light-grey;
  border-radius: 30px;
  font-size: 1em;
  font-weight: bold;
  cursor: pointer;
}

.button__sign-out:hover {
  background: $sign-out-btn-hover;
  border:2px solid $dark-grey;
  color: $red-font;
}

.main__div {
  align-items: center;
  height: 20vh;
  background: rgba(51, 51, 51, .7);
  display: flex;
  justify-content: space-around;
  width: 100%;
}

@keyframes neon4 {
  from {
    box-shadow: 0 0 0px #B6FF00, 0 0 10px #B6FF00, 0 0 30px #fff, 0 0 30px #B6FF00, 0 0 40px #B6FF00, 0 0 60px #B6FF00, 0 0 80px #B6FF00, 0 0 100px #B6FF00;
  }
  to {
    box-shadow: 0 0 10px #fff, 0 0 10px #fff, 0 0 5px #fff, 0 0 20px rgb(255, 44, 44), 0 0 40px rgb(255, 44, 44), 0 0 40px rgb(255, 44, 44), 0 0 60px rgb(255, 44, 44), 0 0 80px rgb(255, 44, 44);
  }
}

form {
  -webkit-animation: neon4 1.5s ease-in-out infinite alternate;
  -moz-animation: neon4 1.5s ease-in-out infinite alternate;
  animation: neon4 1.5s ease-in-out infinite alternate;
  border-radius: 20px;
}


