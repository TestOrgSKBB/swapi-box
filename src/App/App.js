import React, { Component } from 'react';
import Form from '../Form/Form';
import './App.scss'
import { format } from 'url';
import { Route, NavLink } from 'react-router-dom';

import CharacterContainer from '../CharacterContainer/CharacterContainer'
import MovieContainer from '../MovieContainer/MovieContainer'

export default class App extends Component {
  constructor() {
    super()
    this.state= {
      movies: [],
      characters: []
    }
  }



  render() {
    return (
    <body>
        <main>
        <Route exact path='/' component={Form} />
        </main> */}

      </body>
    )
  }
} 