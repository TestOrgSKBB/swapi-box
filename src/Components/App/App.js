import React, { Component } from 'react';
import Form from '../Form/Form';
import './App.scss'
import { Route, NavLink } from 'react-router-dom';
import { fetchData } from '../../apiCalls';
import FavoriteCharactersContainer from '../FavoriteCharactersContainer/FavoriteCharactersContainer';
import CharacterContainer from '../CharacterContainer/CharacterContainer'
import MovieContainer from '../MovieContainer/MovieContainer';
import PropTypes from 'prop-types';

class App extends Component {
  constructor() {
    super()
    this.state= {
      movies: [],
      isLoading: true,
      name: '',
      quote: '',
      rank: '',
      currentMovie: null,
    }
  };

  componentDidMount = () => {
    if(localStorage.getItem('movies')) {
      let movies = localStorage.getItem('movies');
      movies = JSON.parse(movies);
      this.setState({ movies, isLoading: false });
    } else {
      fetchData()
        .then(data => {
          return data;
        })
        .then(movies => {
          this.setState({ movies , isLoading: false})
          const stringyMovies = JSON.stringify(movies);
          localStorage.setItem('movies', stringyMovies);
        })
        .catch(error => console.log(error));
    }
  };

  updateState = (statesObj) => {
    this.setState(statesObj);
  }

  // sortMovies = (movies) => {
  //   const sortedMovies = movies.sort((a, b) => {
  //     return a.episode_id - b.episode_id
  //   })
  //   console.log('movies in app', movies)
  //   console.log('sorted in app', sortedMovies)
  //   this.setState({movies: sortedMovies})
  // };

  selectMovie = e => {
    const episode_id = e.target.closest('section').id;
    this.setState({ currentMovie: episode_id });;
  };

  updateFavorite = name => {
    const movies = this.state.movies.map( movie => movie );
    const movie = movies.find( movie => movie.episode_id === Number(this.state.currentMovie));
    const character = movie.characters.find( character => character.name === name);
    const favorited = character.isFavorited;
    character.isFavorited = !favorited;
    this.setState({ movies });
    localStorage.setItem('movies', JSON.stringify(movies));
  };

  returnFavoriteCharacters = () => {
    const movie = this.state.movies.find(movie => movie.episode_id === Number(this.state.currentMovie));
    const favCharacters = movie.characters.filter(character => character.isFavorited);
    return favCharacters;
  };

  signOut = () => {
    this.setState({ name: '', quote: '', rank: '' });
  };

  render() {
    return (
      <main>
        <div className="main__div">
        <div className="main__div--info">
          <h1 className="user-info user-name">{this.state.name}</h1>
          <h1 className="user-info user-quote">{this.state.quote}</h1>
          <h1 className="user-info user-rank">{this.state.rank}</h1>
        </div>
        <h1 className="title">Star Wars: Trivia</h1>
        <NavLink to='/'>    
          <button className="button__sign-out" onClick={this.signOut}>Sign Out</button>
        </NavLink>
        </div>
        {this.state.isLoading && <h1>Loading...</h1>}
        {!this.state.isLoading && <Route exact path='/' render={() => <Form movies={this.state.movies} updateState={this.updateState} />} />}
        <Route exact path='/movies' render={() => <MovieContainer movies={this.state.movies} selectMovie={this.selectMovie} sortMovies={this.sortMovies} />} />
        <Route exact path='/movies/:id/characters' render={({match}) => {
        const { id } = match.params;
        const opening_crawl = this.state.movies.find(movie => movie.episode_id === parseInt(id)).opening_crawl;
        const characters = this.state.movies.find(movie => movie.episode_id === parseInt(id)).characters;
        const numFavorites = this.returnFavoriteCharacters().length;
        return (<CharacterContainer 
          opening_crawl={opening_crawl} 
          characters={characters} 
          updateFavorite={this.updateFavorite}
          numFavorites={numFavorites} 
        />);
      }} />
        <Route exact path='/movies/favorite-characters' render={() => {
          return <FavoriteCharactersContainer 
            characters={this.returnFavoriteCharacters()} 
            updateFavorite={this.updateFavorite} 
          />
        }} />
    </main>
    );
  };
};

App.propTypes = {
  movies: PropTypes.array,
  isLoading: PropTypes.bool,
  name: PropTypes.string,
  quote: PropTypes.string,
  rank: PropTypes.string,
  sortMovies: PropTypes.func,
  selectMovie: PropTypes.func,
  updateFavorite: PropTypes.func,
  returnFavoriteCharacters: PropTypes.func,
  signOut: PropTypes.func,
}

export default App;

