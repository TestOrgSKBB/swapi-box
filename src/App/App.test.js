import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;
  const mockSignOut = jest.fn()


  beforeEach(() => {
    wrapper = shallow(<App />);
  })

  it('should match the snapshot', () => {
  expect(wrapper).toMatchSnapshot();
  });

  // it('should call signOut on click', () => {
  //   wrapper.find('.button__sign-out').simulate('click');
  //   expect(mockSignOut).toHaveBeenCalled()
  // });

    it('should update the state', () => {
      wrapper.instance().updateState({name: 'obi wan'});
      expect(wrapper.state('name')).toEqual('obi wan');
    });
  })



