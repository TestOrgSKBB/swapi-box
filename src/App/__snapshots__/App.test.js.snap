// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main>
  <div
    className="main__div"
  >
    <h1
      className="title"
    >
      Star Wars: Trivia
    </h1>
    <div
      className="main__div--info"
    >
      <h1
        className="user-info user-name"
      >
        Luke
      </h1>
      <h1
        className="user-info user-quote"
      >
        Star Wars
      </h1>
      <h1
        className="user-info user-rank"
      >
        master
      </h1>
    </div>
    <NavLink
      to="/"
    >
      <button
        className="button__sign-out"
        onClick={[Function]}
      >
        Sign Out
      </button>
    </NavLink>
  </div>
  <h1>
    Loading...
  </h1>
  <Route
    exact={true}
    path="/movies"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:id/characters"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/favorite-characters"
    render={[Function]}
  />
</main>
`;
